### PR TITLE
base interface of the ssd1306 with commands

### PR DESCRIPTION
**Pull request by florianhumblot**

### CODE REVIEW

File: code/Makefile [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #############################################################################
 
 # source files in this project (main.cpp is automatically assumed)
-SOURCES := $(wildcard src/*.cpp)
+SOURCES := $(wildcard src/*.cpp) ../../../modules/i2c-library/code/src/i2c_bus.cpp
```
Comment (gerritvanos) [inline code/Makefile]: this path is incorrect so it seems this should be moved to the libraries folder

Comment (florianhumblot) [reply code/Makefile]: Will be fixed as soon as i2c has been merged, that way I'll be able to thoroughly test the new makefiles

Comment (gerritvanos) [reply code/Makefile]: yes in theory you sould be able to use the include method we discussed:

Comment (florianhumblot) [reply code/Makefile]: This has now been fixed.

File: code/headers/ssd1306.hpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        right_horizontal_scroll = 0x26,
+        left_horizontal_scroll = 0x27,
+        vertical_and_right_horizontal_scroll = 0x29,
+        vertical_and_left_horizontal_scroll = 0x2A
+    };
+
+    /// value to send over i2c before a command
+    constexpr uint8_t ssd1306_cmd_prefix = 0x80;
+
+    /// value to send over i2c before a command
```
Comment (gerritvanos) [inline code/headers/ssd1306.hpp]: i belive this comment should be "before data"?

File: code/main.cpp [hunk 3]
```diff
@@ -1,9 +1,30 @@
...
+#include <i2c_bus.hpp>
+#include <ssd1306_oled_unbuffered.hpp>
 int main(void) {
     // kill the watchdog
     WDT->WDT_MR = WDT_MR_WDDIS;
     // this starts the sysclock
-    hwlib::wait_ms(0);
+    hwlib::wait_ms(20);
+    auto scl = hwlib::target::pin_oc{hwlib::target::pins::scl};
+    auto sda = hwlib::target::pin_oc{hwlib::target::pins::sda};
```
Comment (gerritvanos) [inline code/main.cpp]: i think you don't need these pins anymore since you are using hardware i2c?

**Comment by florianhumblot** [comment]
@gerritvanos I've addressed your comments in the latest commit.

**Pull request merged**